cmd/keeper/keeper.go

```diff
@@ -541,12 +542,22 @@ func (p *PostgresKeeper) Start() {
 	smTimerCh := time.NewTimer(0).C
 	updatePGStateTimerCh := time.NewTimer(0).C
 	publishCh := time.NewTimer(0).C
+	exitSignals := make(chan os.Signal, 1)
+	signal.Notify(exitSignals, syscall.SIGTERM, syscall.SIGINT, syscall.SIGQUIT)
+
 	for true {
 		select {
+		case sig := <-exitSignals:
+			log.Infof("Caught signal: %v.", sig)
+			log.Debugf("Stopping stolon keeper.")
+			cancel()
+			p.pgm.Stop(false)
+			p.end <- nil
```

Comment:
`close(p.end)` assuming this goroutine controls the channel will have the same result and relay the intention of cleanup better.
nit: Change `stopping stolon keeper` -> `Stopping stolon keeper` below to keep log messages consistent.